### ISSUE TITLE
Gradle creates debug APK

### ISSUE THREAD

**Issue by garrettmaring** [fastlane/fastlane#17632]
If you read the rest of the thread, the most common fix is to manually upload an apk yourself once. After that, fastlane should pick it up. 

> On 2. Aug 2019, at 12:37, Ritesh Aryal <notifications@github.com> wrote:
> 
> Has this problem been resolved yet ?
> Please suggest.
> 
> My usecase is:
> 
> using fastlane to upload my apk directly to Play Store
> m getting that error the second after I apply fastlane supply init command
> And I'm not sure why is it not happy, even if I've provided the correct package name (which is same as package_id, if I'm not mistaking)
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.

_Originally posted by @lazerwalker in https://github.com/fastlane/fastlane/issues/14686#issuecomment-517660458_

**Comment by garrettmaring** [comment]
I am trying to upload to Google Play. For the first upload, it's failing on package name and the recommendation from #14686 is to upload an APK manually.

I've then run a assembleRelease but am seeing this error when dragging it into the Google Play store.

```
  lane :beta do
    gradle(
      task: 'assemble',
      build_type: 'Release'
    )
  end
```

![Screen Shot 2020-11-18 at 12 09 40 PM](https://user-images.githubusercontent.com/8270120/99582586-2c99a780-2997-11eb-807e-c1cdea8bf365.png)

**Comment by max-ott** [comment]
As far as I know the Google Play Publishing API, which `fastlane` relies on, requires you to do the initial setup manually on the Google Play Dev Console. With regards to `gradle`, I recommend checking out their docs and see why you are creating a debug APK.

**Comment by garrettmaring** [comment]
@max-ott it does require initial setup through Google Play but that setup requires an initial AAB or APK to be uploaded. This upload requires a release + signed build. Eventually, I just followed https://reactnative.dev/docs/next/signed-apk-android to build the first bundle signed in release mode. With `supply`, I assumed that the signing process was done by fastlane.

**Comment by joshdholtz** [comment]
@garrettmaring The `gradle` action is just a wrapper for the `gradlew` CLI command. _fastlane_ does not do any signing with either `gradle` or `supply`. `supply` is the layer that uploads a signed APK to Google Play.

It seems like there might be something missing in your gradle file the is causing `assembleRelease` to not get signed. If you run the CLI command `./gradlew assembleRelease` in your project, it should produce the same debug APK that _fastlane_ is building with the `gradle` action.

You might need to do something like this in your gradle file - https://stackoverflow.com/a/21020469

Hope this helps!

**Comment by anh1979** [comment]
I think I encountered this myself, but not 100% sure:
We had an order version of gradle installed and assembleRelease DID put out a signed build. When we switched our system over to use gradle version >= 6.5, assembleRelease did suddenly put out a debug apk instead. 

Maybe we could have avoided that by changing the gradle configuration and build.properties - but I didn't want to change that due to not knowing enough about that.

I instead do a build myself by command line (it is an ionic cordova app) and then use "assemble" in my gradle action instead of "assembleRelease". 

That worked out for me - maybe it helps you, too.
